android/KMEA/app/src/main/java/com/keyman/engine/KMManager.java

```diff
@@ -2218,20 +2218,45 @@ public static int getBannerHeight(Context context) {
   }
 
   public static int getKeyboardHeight(Context context) {
+    int orientation = getOrientation(context); // Default behavior
+    return getKeyboardHeight(context, orientation); // Delegate to the overloaded method
+  }
+
+  // Overloaded method with an additional orientation parameter
+  public static int getKeyboardHeight(Context context, int orientation) {
     int defaultHeight = (int) context.getResources().getDimension(R.dimen.keyboard_height);
     SharedPreferences prefs = context.getSharedPreferences(context.getString(R.string.kma_prefs_name), Context.MODE_PRIVATE);
-
-    int orientation = getOrientation(context);
     if (orientation == Configuration.ORIENTATION_PORTRAIT) {
-      return prefs.getInt(KMManager.KMKey_KeyboardHeightPortrait, defaultHeight);
+        return prefs.getInt(KMManager.KMKey_KeyboardHeightPortrait, defaultHeight);
     } else if (orientation == Configuration.ORIENTATION_LANDSCAPE) {
-      return prefs.getInt(KMManager.KMKey_KeyboardHeightLandscape, defaultHeight);
+        return prefs.getInt(KMManager.KMKey_KeyboardHeightLandscape, defaultHeight);
     }
-
     return defaultHeight;
   }
 
   public static void applyKeyboardHeight(Context context, int height) {
+    SharedPreferences prefs = context.getSharedPreferences("KMAPreferences", Context.MODE_PRIVATE);
+    SharedPreferences.Editor editor = prefs.edit();
+    int orientation = getOrientation(context);
+    if (height == 0) {
+      // Passing 0 will reset both stored heights
+      editor.putInt(KMManager.KMKey_KeyboardHeightLandscape, getKeyboardHeight(context, Configuration.ORIENTATION_LANDSCAPE));
+      editor.putInt(KMManager.KMKey_KeyboardHeightPortrait, getKeyboardHeight(context, Configuration.ORIENTATION_PORTRAIT));
+      editor.commit();
+    
+      int height = (orientation == Configuration.ORIENTATION_PORTRAIT) ? getKeyboardHeight(context, Configuration.ORIENTATION_PORTRAIT) : getKeyboardHeight(context, Configuration.ORIENTATION_LANDSCAPE);
+    }
+    else: {
```

Comment:
`height` is already passed into the function. Also fixing a stray `:` syntax
```suggestion

    } else {
```

Reply:
Yes, but in this context, `height` has been passed as `0` (the reset command). The deleted line gets the default height and overrides `height` with the default height before running your layout code after leaving the if. I've pushed what should be the right version.

Reply:
OK, I've deferred to you and Marc. The deleted line is now gone.